staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -240,7 +244,7 @@ func (s *DelegatingAuthenticationOptions) getClientCA() (*ClientCertAuthenticati
 		return nil, err
 	}
 	if incluster == nil {
-		return nil, fmt.Errorf("cluster doesn't provide client-ca-file")
+		return &s.ClientCert, IgnorableError("cluster doesn't provide client-ca-file in configmap/%s in %s, so client certificate authentication to extension api-server won't work.", authenticationConfigMapName, authenticationConfigMapNamespace)
```

Comment:
could live with this, but don't think it's very pretty.

For some bike-shedding: make the return type `cert, error, error` with the first error to be just a warning, the last one to be fatal.

Reply:
I was using the recommendation from Error handling [blog](https://blog.golang.org/error-handling-and-go) to define a custom error type and use type assertion. This is also done in the certificates controller code of Kubernetes.

I think return order of two errors dictating client behavior may be more fragile.